semigroups/semifp.py

```diff
@@ -64,7 +199,18 @@ def __init__(self, alphabet, rels):
             raise TypeError('the second argument (relations) must be a ' +
                             'list of pairs of strings')
 
-        # TODO parse relations using ^ etc here, as per in ParseRelators in GAP
+        pure_letter_alphabet = True
+        for i in alphabet:
```

Comment:
Right, I guess my point was about the "assumed" part, should you not check that it is already parsed?